src/Microsoft.DocAsCode.Build.Engine/HostService.cs

```diff
@@ -123,6 +124,43 @@ public MarkupResult Markup(string markdown, FileAndType ft, bool omitParse)
             return MarkupCore(markdown, ft, omitParse);
         }
 
+        #region TableOfContent related
+
+        public ImmutableArray<RestructureTableOfContent> RestructureTableOfContentDelegates { get; set; } = ImmutableArray<RestructureTableOfContent>.Empty;
+
+        public bool ShouldRestructureTableOfContent()
+        {
+            return RestructureTableOfContentDelegates.Length > 0;
+        }
+
+        public void RegisterRestructuringTableOfContent(params RestructureTableOfContent[] methods)
+        {
+            if (methods == null || methods.Length == 0)
+            {
+                return;
+            }
+
+            var list = new List<RestructureTableOfContent>(methods);
+            lock (_tocSyncRoot)
+            {
+                if (RestructureTableOfContentDelegates != null)
```

Comment:
get it, so it's better also check null in ShouldRestructureTableOfContent and InvokeRestructuringTableOfContent?

---
In reply to: [93891813](https://github.com/dotnet/docfx/pull/1059#discussion_r93891813) [](ancestors = 93891813)